fdbclient/NativeAPI.actor.cpp

```diff
@@ -9886,6 +9886,37 @@ Reference<DatabaseContext::TransactionT> DatabaseContext::createTransaction() {
 }
 
 // BlobGranule API.
+ACTOR Future<Standalone<VectorRef<KeyRangeRef>>> getBlobRanges(Transaction* tr, KeyRange range, int batchLimit) {
+	state Standalone<VectorRef<KeyRangeRef>> blobRanges;
+	state Key beginKey = range.begin;
+
+	loop {
```

Comment:
it is generally bad practice to have a retry loop inside a function that takes a pointer to a transaction. If a transaction uses this function along with other operations on the transaction, those other operations will not be retried.